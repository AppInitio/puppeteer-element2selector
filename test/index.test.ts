import { element2selector, Options } from '../';

beforeAll(async () => {
  await page.goto('http://localhost:1234/simple');
});

test.each<[string, Options | undefined]>([
  ['h1', undefined],
  ['article:nth-of-type(2) > section > h2', undefined],
  ['.blog1 p:nth-of-type(3)', undefined],
  ['#note', undefined],
  ['#note', { seedMinLength: 2 }],
])('simple - %s - %p', async (input, options) => {
  const element = await page.$(input);
  expect(await element2selector(element!, options)).toMatchSnapshot();
});
