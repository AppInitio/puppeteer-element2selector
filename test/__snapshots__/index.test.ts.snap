// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`simple - #note - {"seedMinLength": 2} 1`] = `"section > #note"`;

exports[`simple - #note - undefined 1`] = `"#note"`;

exports[`simple - .blog1 p:nth-of-type(3) - undefined 1`] = `".blog1 > section:nth-child(1) > p:nth-child(4)"`;

exports[`simple - article:nth-of-type(2) > section > h2 - undefined 1`] = `".blog2 > section:nth-child(1) > h2"`;

exports[`simple - h1 - undefined 1`] = `"h1"`;
